scripts/build-shrink-ray.js

```diff
@@ -1,73 +1,95 @@
-/* eslint-disable no-console */
-
-require('isomorphic-fetch');
-
 const {resolve} = require('path');
 const {execSync} = require('child_process');
-const {postReport, sendCommitStatus} = require('@shopify/shrink-ray');
-
-const BASE_BRANCH = 'master';
-const repo = 'polaris-react';
-const sha = process.env.TRAVIS_COMMIT;
-
-const report = resolve(
-  __dirname,
-  '../build/storybook/bundle-analysis/report.html',
-);
+const {readFile} = require('fs-extra');
+const {Logger, Build, ShrinkRayAPI, Check} = require('@shopify/shrink-ray');
 
 process.on('unhandledRejection', (reason) => {
   throw reason;
 });
 
-if (sha) {
-  setupShrinkRay();
-} else {
-  console.log(
-    'sha is not available, building bundle without pinging shrink-ray',
-  );
-  buildPackages();
-}
+startShrinkRayBuild({
+  masterBranchName: 'master',
+  repo: 'polaris-react',
+  sha: process.env.TRAVIS_COMMIT,
+  reportPath: resolve(
+    __dirname,
+    '..',
+    'build/storybook/bundle-analysis/report.html',
+  ),
+  buildUrl: process.env.TRAVIS_JOB_WEB_URL,
+  skip: [Check.Entrypoints],
+});
 
-function buildPackages() {
-  execSync('yarn run storybook:build --quiet', {
-    stdio: 'inherit',
-  });
-}
+async function startShrinkRayBuild({
+  masterBranchName,
+  repo,
+  sha,
+  reportPath,
+  buildUrl,
+  skip,
+}) {
+  const logger = new Logger();
+
+  if (sha) {
```

Comment:
I've never needed to run this without having the sha and thus wanting to run just the build - i'd just use `yarn run storybook:build` directly.

Let's change this so we exit early with an error message if we haven't got a sha. Then we can inline the contents of `buildStorybookBuild` into this function